### ISSUE TITLE
After Upgrade VS Code isn't hitting breakpoints anymore (using webpack)

### ISSUE THREAD

**Issue by radomeit** [Microsoft/vscode-chrome-debug#249]
Hi, I'm using 
Angular2-Webpack-Starter and VS Code 1.6 with Debugger for Chrome 0.5.2.
If I upgrade to Debugger for Chrome 1.1 the breakpoints in my app's .ts-files aren't being hit anymore.
After downgrading to 0.5.2 they start working again, so it is probably an issue with the chrome debugger and not with webpack / the sourcemaps.

According to the diagnostic logs the debugger uses the webpack generated
localhost:3000\main.map which contains the correct source maps.

But there is a difference in the diagnostic logs of 0.5.2 and 1.1.:

0.5.2 contains lines like these (mapping the ts files I want to debug):
SourceMap: mapping webpack:///./src/app/dashboard/dashboard.module.ts => c:\<...>\app\src\app\dashboard\dashboard.module.ts, via sourceMapPathOverrides entry - "webpack:///*": "c:\<...>\app/*"

1.1. simply doesn't. It seems like the new version simply ignores these files. I haven't found any errors in the logs which could give a hint why this happens.

my launch.json:
{
    "version": "1.0.0",
    "configurations": [
        //Launch Chrome with debugging enabled
        {
            "name": "Launch Chrome against localhost, with sourcemaps",
            "type": "chrome",
            "request": "launch",
            "url": "http://localhost:3000/*",
            "runtimeArgs": [
                "--disable-web-security",
                "--user-data-dir",
                "--remote-debugging-port=9222"
            ],
            "sourceMaps": true
        },
        //Attach Debugger to Chrome
        {
            "name": "Attach to Chrome, with sourcemaps",
            "type": "chrome",
            "request": "attach",
            "url": "http://localhost:3000/*",
            "port": 9222,
            "sourceMaps": true,
            "webRoot": "${workspaceRoot}",
            "diagnosticLogging": true
        }
    ]
}

Any clues?

**Comment by JonKragh** [comment]
I had a bunch of issues with the 1.1.0 extension and VS code 1.6.

I downgraded to extension 0.5.2 and also added this flag to my launch config:

"userDataDir": "C:\\temp\\chromeDummyDir"

Documented here:

https://github.com/Microsoft/vscode-chrome-debug/issues/116

Now I am hitting breakpoints in VS Code attached to chrome.

**Comment by roblourens** [comment]
@radomeit Could you try adding this manually to your launch config? `"sourceMapPathOverrides": { 'webpack:///*': '${webRoot}/*' }`. That rule should be applied by default, but I did move the code that applies it. Not sure why it's getting lost. If that doesn't work, it would be helpful to see your log.

**Comment by roblourens** [comment]
@JonKragh What other issues did you see with 1.1.0? Please file issues! :)

**Comment by JonKragh** [comment]
@roblourens I kept getting that can't connect issue that seems to have been closed a few days ago.  I will attempt to repro issues tomorrow and report back. I had to finish a project for work so once I got it to work I needed to debug.  Thank you for this debugger it rules!

**Comment by radomeit** [comment]
@roblourens I've added 
            "sourceMapPathOverrides": {
                "webpack:///*": "${webRoot}/*"
            },
to both sections in the configurations array in launch.json.
In 1.1 it's still not working with the added sourceMapPathOverrides. 0.5.2 works always (with or without the added sourceMapPathOverrides).

I've attached a snippet from the comparison of the logs of the two versions. V1.1 seems to have a problem handling the sourcemap for main.bundle.js (=my app's code) while 0.5.2 hasn't.

[Comparison_1_1_with_0_5_2_log.txt](https://github.com/Microsoft/vscode-chrome-debug/files/523786/Comparison_1_1_with_0_5_2_log.txt)

**Comment by skolmer** [comment]
Are sourcemaps working in chrome dev tools? I faced the same issues with latest chrome versions and had to remove the devtool setting from my webpack config an added SourceMapDevToolPlugin() instead.
https://github.com/webpack/webpack/issues/2145

**Comment by roblourens** [comment]
@radomeit One thing is that in 1.1, it says that main.bundle.js only has one source -

```
SourceMap: creating for http://localhost:3000/main.bundle.js
SourceMap: sourceRoot: 
SourceMap: sources: ["webpack:///./src/app/dashboard/dashboard.item.readonly.component.html?c570"]
```

and that should be dumped directly from the sourcemap. But 0.5.2 has the same line but with a bunch of sources. I'm not sure how that would go wrong on my end. I assume you're connecting to exactly the same app though. Can you check that it all shows up in chrome devtools?

**Comment by radomeit** [comment]
@roblourens Yes, I'm connecting to exactly the same app with both versions of chrome debugger.
Debugging the .ts - files in chrome devtools is no problem, they are shown under webpack://./src/app

@skolmer I removed the "devtool: 'cheap-module-source-map'" setting in my webpack.dev.js and instead added the SourceMapDevToolPlugin as you suggested and that fixed it for me in 1.1 :) Many thanks! (0.5.2 and chrome dev tools still work as well btw)
